executor/join.go

```diff
@@ -529,6 +558,36 @@ func (e *NestedLoopApplyExec) prepare(goCtx goctx.Context) error {
 	}
 }
 
+// prepare4Chunk reads all data from the small table and stores them in a slice.
+func (e *NestedLoopApplyExec) prepare4Chunk(goCtx goctx.Context) error {
+	err := e.SmallExec.Open(goctx.TODO())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer terror.Call(e.SmallExec.Close)
+	var selected []bool
```

Comment:
this slice is allocated every time a new outer row is fetched, we'd better buffer it outside this function.